quickjs/src/commonMain/kotlin/app/cash/quickjs/KtBridge.kt

```diff
@@ -15,21 +15,42 @@
  */
 package app.cash.quickjs
 
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
 // TODO(jwilson): merge with QuickJs?
 class KtBridge internal constructor(
   private val outboundBridge: InternalBridge
 ) {
   private val inboundHandlers = mutableMapOf<String, InboundService<*>>()
+  private var nextId = 1
 
   internal val inboundBridge = object : InternalBridge {
-    override fun invokeJs(
+    override fun invoke(
       instanceName: String,
       funName: String,
       encodedArguments: ByteArray
     ): ByteArray {
       val handler = inboundHandlers[instanceName] ?: error("no handler for $instanceName")
       return handler.call(InboundCall(funName, encodedArguments, handler.jsAdapter))
     }
+
+    override fun invokeSuspending(
+      instanceName: String,
+      funName: String,
+      encodedArguments: ByteArray,
+      callbackName: String
+    ) {
+      val handler = inboundHandlers[instanceName] ?: error("no handler for $instanceName")
+      // TODO(jwilson): confirm how we bootstrap a coroutine on an inbound call.
+      GlobalScope.launch {
```

Comment:
This is almost certainly wrong but I don’t know what’s right.

Reply:
We will likely have to accept a CoroutineScope when creating the QuickJs instance.